crates/strategies/blend-strategies/src/auction_manager.rs

```diff
@@ -225,6 +226,113 @@ impl OngoingAuction {
         profit
     }
+    pub fn build_requests(
+        &self,
+        wallet: &HashMap<String, i128>,
+        pool_position: &UserPositions,
+        supported_collateral: &Vec<String>,
+        min_hf: &i128,
+    ) -> Result<Vec<Request>> {
+        let mut new_pool_positions = pool_position.clone();
+        let mut requests: Vec<Request> = vec![Request {
+            request_type: 6 + self.auction_type,
+            address: self.user.clone(),
+            amount: self.pct_to_fill as i128,
+        }];
+
+        if self.auction_type == 0 || self.auction_type == 1 {
+            for (bid_asset, bid_value) in self.auction_data.bid.clone() {
+                if wallet.get(&bid_asset).is_some()
+                    && wallet[&bid_asset] > 100
+                    && self.target_block - self.auction_data.block < 400
+                {
+                    let reserve = self
+                        .db_manager
+                        .get_reserve_config_from_asset(&self.pool, &bid_asset)?;
+                    let wallet_balance = wallet.get(&bid_asset).unwrap();
+                    requests.push(Request {
+                        request_type: 5,
+                        address: bid_asset.clone(),
+                        amount: *wallet_balance,
+                    });
+                    let wallet_dtoken_bal = wallet_balance
+                        .fixed_div_floor(SCALAR_9, reserve.est_d_rate)
+                        .unwrap();
+                    if wallet_dtoken_bal < bid_value {
+                        if new_pool_positions.liabilities.contains_key(&bid_asset) {
+                            new_pool_positions
+                                .liabilities
+                                .entry(bid_asset.clone())
+                                .and_modify(|e| *e += bid_value - wallet_dtoken_bal);
+                        } else {
+                            new_pool_positions
+                                .liabilities
+                                .insert(bid_asset.clone(), bid_value - wallet_dtoken_bal);
+                        }
+                    }
+                }
+            }
+        }
+        let mut liquidator_effective_collateral = sum_adj_asset_values(
+            new_pool_positions.collateral.clone(),
+            &self.pool,
+            true,
+            &self.db_manager,
+        )
+        .unwrap()
+        .1;
+        let liquidator_effective_liabilities = sum_adj_asset_values(
+            new_pool_positions.liabilities.clone(),
+            &self.pool,
+            false,
+            &self.db_manager,
+        )
+        .unwrap()
+        .1;
+        if self.auction_type == 0 {
+            for (index, asset) in supported_collateral.iter().enumerate() {
+                if index != 0 {
+                    let reserve = self
+                        .db_manager
+                        .get_reserve_config_from_asset(&self.pool, asset)?;
```

Comment:
(picky) we should probably move this DB call to after the `b_tokens.is_some()` check